### ISSUE TITLE
Inconsistent Output for --output-type plist

### ISSUE THREAD

**Issue by laus102** [FelixHerrmann/swift-package-list#124]
We're trying to generate a secondary `Acknowledgements-spm.plist` file in our `.xcworkspace` because we are currently still using Cocoapods to generate an `Acknowledgements.plist`.  

This way, while we are still in the process of moving our dependencies from Cocoapods to SPM, we can still display accurate acknowledgements for both managers (and avoid getting sued).

| Real Output  | Desired Output |
| ------------- | ------------- |
| ![image](https://github.com/user-attachments/assets/b5f38867-d42f-4a7c-bba2-a2ffbd61774b)  | ![image](https://github.com/user-attachments/assets/ccf1ee97-6373-4251-86d6-dc3c321d21df)  |

The output on the right is what I believe `swift-package-list` generated one time, however I have yet to be able to replicate these same results, as now the tool only produces `.plist` files of the format on the left.  

The file on the left produces a blank Acknowledgements page in the iOS Settings page for the app. 

The file on the right produces a functional Acknowledgements page.

Most notably, the file on the right features the key `PreferenceSpecifiers` while the file on the left doesn't.

Am I missing a command line argument, or some other "gotcha"? 

## The command we're using: 

```
swift-package-list 
	[...].xcworkspace 
	--output-type plist 
	--custom-file-name Acknowledgements-spm-1 
	--output-path [...]/SupportingFiles/Settings.bundle
```

## The file structure:

```
/ Settings.bundle
	Acknowledgements-spm.plist
	Acknowledgements.plist
	Root.plist
````

**Comment by FelixHerrmann** [comment]
Thanks for reaching out! I already see the problem, using the `plist` output will just generate a plain property list from the array of packages equivalent to the `json` output, this will not work with a settings bundle because this requires specific keys; what you want is the `settings-bundle` output, if not already present it will created the necessary folder/file structure of a `Settings.bundle` (including localization), if it is already present this tool only "owns" `Acknowledgements.plist` and the `Packages` subdirectory.

Based on what you've send this seems to overlap with what you get generated from Cocoapods, which tool are you using and can you configure the filename there?

**Comment by laus102** [comment]
Hi, i'm taking your advice and using the `settings-bundle` output, utilizing the build tool plugin. 

I am receiving this error when trying to build to iOS.  
![image](https://github.com/user-attachments/assets/0f48bb54-4cf5-4d68-be03-2b97719c6f02)

Any tips?

**Comment by FelixHerrmann** [comment]
Pretty sure you've added the executable to your app target, just remove that there. But in your special case I think you have to generate the bundle manually so you know where it is to add the plist from cocoapods to it. Let me know if you need further guidance, I'm confident we will get this dual setup working for you!

**Comment by laus102** [comment]
Ok one more...

![image](https://github.com/user-attachments/assets/3bf90877-0200-42c9-8ccc-45f6df8f98ee)
![image](https://github.com/user-attachments/assets/1b0eed6b-6611-46ca-a6e0-b90e9f35c2c5)

But i'm getting the following error: 

![image](https://github.com/user-attachments/assets/7b52432f-52dc-4868-9e7c-82ba19a79b36)

**Comment by FelixHerrmann** [comment]
I see the issue, because you are using Cocoapods you have to use a workspace. Once a project is part of a workspace the `Package.resolved` is located in the *.xcworkspace file. So to tell the plugin what to do in that non-default scenario you have to create a `swift-package-list-config.json` file in the root of the project, as explained in the [Build Tool Plugin section](https://github.com/FelixHerrmann/swift-package-list?tab=readme-ov-file#build-tool-plugin); in your case the following config should be sufficient:
```json
{
    "projectPath" : "YourWorkspaceName.xcworkspace"
}
```

But, as mentioned before I don't think we will be able to achieve a mixed Cocoapods/SPM solution with a SPM plugin because the output is somewhere deep inside of the `DerivedData` directory; manual invocation or build script the generates the `Settings.bundle` into your project file structure is what I would do. If you can explain me how you generate the Cocoapods acknowledgment file I can try to build that on my side and provide an example repo or something.

**Comment by laus102** [comment]
@FelixHerrmann 
![Screenshot 2024-10-07 at 4 43 05 PM](https://github.com/user-attachments/assets/2d732765-8060-4cc1-8665-a8a0c438f22e)

My strategy right now is to create a separate `.plist` file within the Settings bundle, one for Cocoapods, and one for SPM.

Not the prettiest, but at least we won't get sued 🤷 

If you have any better ideas i'm all ears 😂

**Comment by laus102** [comment]
![image](https://github.com/user-attachments/assets/a2e2786b-9f9e-44d8-a046-a55b7e1da61e)

With the Cocoapods `post_install` looking like this:

```
post_install do | installer |
    require 'fileutils'
    FileUtils.cp_r('Pods/Target Support Files/Pods-XXXX/Pods-XXXX-acknowledgements.plist', 'XXXX/XXXX/SupportingFiles/Settings.bundle/Acknowledgements-cp.plist', :remove_destination => true)
```

**Comment by FelixHerrmann** [comment]
Yep that is exactly the way you have to combine it; looks like you are free on the Cocoapods side in terms of file naming which is good.

> If you have any better ideas i'm all ears 😂
From what you've send I cannot tell how you generate the SPM side but Cocoapods looks good.

**Comment by FelixHerrmann** [comment]
Ok, I tried it on my side and it works all as expected. Just to clarify that again, if you want to mix SPM with Cocoapods you have use a `Run Script Phase`, the convenient package plugin will not allow the necessary modifications.

I have created a [example repo](https://github.com/FelixHerrmann/swift-package-list-example-cocoapods) with three branches:
- [master](https://github.com/FelixHerrmann/swift-package-list-example-cocoapods/tree/master) is the starting point of a mixed codebase with SPM and Cocoapods
- [package-plugin](https://github.com/FelixHerrmann/swift-package-list-example-cocoapods/tree/package-plugin) is the plugin version without listing the pods which you cannot use
- [run-script](https://github.com/FelixHerrmann/swift-package-list-example-cocoapods/tree/run-script) is the solution you need

Just pull it and play around with it, my commit messages should be helpful step-by-step guide. Let me know if you have any more questions.

**Comment by laus102** [comment]
Thanks so much for the example repo! 

Couple of questions:

1) I added some additional logging to the `Run Script Phase` but am failing to see anything in the console...
<img width="1629" alt="image" src="https://github.com/user-attachments/assets/bcf2eb8b-670a-4fce-bed9-fcc19544aeab">

2) I am failing to see where / how the `swift-package-list` tool is installed

Thanks so much again!

**Comment by FelixHerrmann** [comment]
No problem, I think this is a helpful guide for anyone in a similar situation.

> 1. I added some additional logging to the Run Script Phase but am failing to see anything in the console...

Console is runtime output, but this is executed during the build-time. You need to head over to the report navigator and select your build there (can also filter for it):

<img width="1963" alt="Screenshot 2024-10-14 at 23 08 44" src="https://github.com/user-attachments/assets/7bfb2745-0993-4501-ade4-dfd530c9422b">

> 2. I am failing to see where / how the swift-package-list tool is installed

If you have a run script you have to install the tool on your machine (and CI environment etc.). Easiest method is Homebrew, see the README section [here](https://github.com/FelixHerrmann/swift-package-list#using-homebrew), the first line in the script (export PATH...) is necessary for that; once installed you can confirm with `swift-package-list --version` or `which swift-package-list` in your Terminal (or via your print in your script).

Let me know if something is still not working or if you have any other question.

**Comment by Malove98** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
